Problem with `test_jsonschemagen.py` unit tests

**Describe the bug**

When building the test environment, I found that `tests/test_generators/test_jsonschemagen.py` tests were failing.  Here is a typical error:
```
file /home/paul/git/linkml/tests/test_generators/test_jsonschemagen.py, line 258
  def test_required_slot_condition_in_rule(subtests, input_path):
E       fixture 'subtests' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, doctest_namespace, input_path, kitchen_sink_path, monkeypatch, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, snapshot, snapshot_path, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.
```

**To reproduce**

Check out `linkml` git repo.
Build virtual environment following instructions
Run `poetry run pytest tests/test_generators/test_jsonschemagen.py`

**Expected behavior**

There should not be any errors reported for this test suite.

**About your computer (if applicable, please complete the following information):**

```console
paul@monkeywrench:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 12 (bookworm)
Release:        12
Codename:       bookworm
paul@monkeywrench:~$ 
```

**Additional context**

The problem is due to a missing dependency.  The `test_jsonschemagen.py` tests depend  on the `pytest-subtests` package being installed.

Manually installing this dependency fixes the problem:

```console
paul@monkeywrench:~$ pip3 install pytest-subtests
Collecting pytest-subtests
  Using cached pytest_subtests-0.11.0-py3-none-any.whl (6.7 kB)
Requirement already satisfied: pytest>=7.0 in ./Venv/test-linkml/lib/python3.11/site-packages (from pytest-subtests) (7.4.4)
Requirement already satisfied: attrs>=19.2.0 in ./Venv/test-linkml/lib/python3.11/site-packages (from pytest-subtests) (23.2.0)
Requirement already satisfied: iniconfig in ./Venv/test-linkml/lib/python3.11/site-packages (from pytest>=7.0->pytest-subtests) (2.0.0)
Requirement already satisfied: packaging in ./Venv/test-linkml/lib/python3.11/site-packages (from pytest>=7.0->pytest-subtests) (23.2)
Requirement already satisfied: pluggy<2.0,>=0.12 in ./Venv/test-linkml/lib/python3.11/site-packages (from pytest>=7.0->pytest-subtests) (1.3.0)
Installing collected packages: pytest-subtests
Successfully installed pytest-subtests-0.11.0
paul@monkeywrench:~$ 
```

I believe I followed the instructions.  If so, then the `pytest-subtests` dependency should either be documented or configuration updated to ensure this dependency is pulled in automatically.

--- a/docs/contributing/contributing.md
+++ b/docs/contributing/contributing.md
@@ -2,6 +2,18 @@
 
 For general info about how to contribute to LinkML, please see [FAQ: Contributing](https://linkml.io/linkml/faq/contributing.html).
 
+### Development Environment Setup
+
+LinkML is developed with [Python](https://www.python.org/) and uses [Poetry](https://python-poetry.org/) for dependency management and packaging. If you have not used Poetry before refer to the [Poetry documentation](https://python-poetry.org/docs/master/#installation/) for installation instructions.
+
+Then clone the repository and install the development dependencies:
+
+```shell
+git clone https://github.com/linkml/linkml
+cd linkml
+poetry install
+```
+
 ### LinkML Testing Framework
 
 * The LinkML test suite can be found in the [tests](../../tests/) folder. The tests are written under specific subfolders within the main aggregated [tests](../../tests/) suite.

